### ISSUE TITLE
normal behavior of GAN

### ISSUE THREAD

**Issue by bobchennan** [soumith/ganhacks#5]
I am wondering is there any rules to check whether GAN converges or not if I cannot check the generators directly. Those tricks are very helpful but when I face the loss it is still hard for me to make conclusion.

So in my case this is the result:
Epoch 1:

```
component              | loss | generation_loss | auxiliary_loss
-----------------------------------------------------------------
generator (train)      | 5.93 | 0.73            | 5.20
generator (test)       | 2.48 | 1.50            | 0.98
discriminator (train)  | 6.22 | 0.59            | 5.63
discriminator (test)   | 4.13 | 0.32            | 3.82

```
Epoch 2:
```
component              | loss | generation_loss | auxiliary_loss
-----------------------------------------------------------------
generator (train)      | 3.06 | 0.88            | 2.18
generator (test)       | 1.66 | 1.51            | 0.14
discriminator (train)  | 4.03 | 0.44            | 3.58
discriminator (test)   | 3.43 | 0.28            | 3.16
```
Epoch 3:
```
component              | loss | generation_loss | auxiliary_loss
-----------------------------------------------------------------
generator (train)      | 2.50 | 0.83            | 1.67
generator (test)       | 1.76 | 1.70            | 0.06
discriminator (train)  | 3.51 | 0.36            | 3.14
discriminator (test)   | 3.20 | 0.20            | 2.99
```
Epoch 4:
```
component              | loss | generation_loss | auxiliary_loss
-----------------------------------------------------------------
generator (train)      | 2.27 | 0.73            | 1.54
generator (test)       | 2.19 | 2.16            | 0.03
discriminator (train)  | 3.27 | 0.29            | 2.98
discriminator (test)   | 3.05 | 0.13            | 2.91
```
Epoch 5:
```
component              | loss | generation_loss | auxiliary_loss
-----------------------------------------------------------------
generator (train)      | 2.06 | 0.62            | 1.44
generator (test)       | 2.68 | 2.66            | 0.02
discriminator (train)  | 3.11 | 0.21            | 2.89
discriminator (test)   | 2.96 | 0.09            | 2.87
```

Loss of generator and discriminator are both decreasing but the loss on cv set is increasing. I already insert noise to the discriminator and used dropout for generator(but I think it is not used for cv).

**Comment by soumith** [comment]
at the moment i am not aware of any work or hacks where we can correlate the losses with sample quality

**Comment by bobchennan** [comment]
So what the only thing we can do is to check the sample quality directly? Because in some cases like [toy example for MNIST](https://github.com/lukedeo/keras-acgan/blob/master/acgan-analysis.ipynb), the discriminator and generator loss will converge to the same number. It looks nice but it is hard to achieve in complex problem.

**Comment by soumith** [comment]
yes the only thing to do is to check sample quality.
There are some other ways of checking, by training N generators-discriminator pairs and cross-validating them against each other. But this also does not absolutely guarantee sample quality correlation with the cross-validation score (for example in the trivial case where all G-D pairs have collapsed)
